release-common-lib/src/main/java/org/reactome/util/compare/DBObjectComparer.java

```diff
@@ -19,135 +19,157 @@
  * This class can be used to perform comparisons on any two DatabaseObjects across two different databases.
  * (It assumes that they have the same DB_ID).
  * @author sshorser
- *
  */
 public class DBObjectComparer
 {
 	/**
 	 * Compares two GKInstances.
-	 * @param inst1 - the first instance.
-	 * @param inst2 - the second instance.
-	 * @param sb - a StringBuilder that will contain a detailed report of differences.
+	 * @param inst1 the first instance.
+	 * @param inst2 the second instance.
+	 * @param sb a StringBuilder that will contain a detailed report of differences.
 	 * @return The number of differences between the two instances.
 	 * A single-valued attribute that differs will count as 1 diff.
 	 * If the instances have different schema classes, that will count as 1 diff.
-	 * If a multi-valued attribute has a different number of elements between the two intances, 
+	 * If a multi-valued attribute has a different number of elements between the two instances,
 	 * that will count as 1 diff and the elements will NOT be compared.
 	 */
 	public static int compareInstances(GKInstance inst1, GKInstance inst2, StringBuilder sb)
 	{
 		return compareInstances(inst1, inst2, sb, 5, false);
 	}
-	
+
 	/**
 	 * Compares two GKInstances.
-	 * @param inst1 - the first instance.
-	 * @param inst2 - the second instance.
-	 * @param sb - a StringBuilder that will contain a detailed report of differences.
-	 * @param checkReferrers - Should referring instances also be checked? If <b>true</b>, then referring attributes will <em>also</em> be checked for differences.
-	 * They will be followed to the same recursion depth as regular attributes. Using this with a high maxRecursionDepth could lead to a very long execution time. Be careful!!
-	 * You might also want to use a custom predicate to filter out attributes that could lead to a cyclical difference check.
+	 * @param inst1 the first instance.
+	 * @param inst2 the second instance.
+	 * @param sb a StringBuilder that will contain a detailed report of differences.
+	 * @param checkReferrers Should referring instances also be checked? If <b>true</b>, then referring attributes
+	 * will <em>also</em> be checked for differences. They will be followed to the same recursion depth as regular
+	 * attributes. Using this with a high maxRecursionDepth could lead to a very long execution time. Be careful!!
+	 * You might also want to use a custom predicate to filter out attributes that could lead to a cyclical difference
```

Comment:
Perhaps for this last statement it would be good to provide an example of an attribute that would differ over releases

Reply:
@SolomonShorser-OICR Could you provide an example of a custom predicate that differs from the default we could include in the JavaDocs?

Reply:
Check to javadoc for the overload of this function that actually takes a Predicate (line 114).

For another example, see the comparison test for GoUpdate: https://github.com/reactome/data-release-pipeline/blob/develop/go-update/src/test/java/org/reactome/release/goupdate/JavaVsPerlComparison.java#L117

and also here: https://github.com/reactome/data-release-pipeline/blob/develop/go-update/src/test/java/org/reactome/release/goupdate/JavaVsPerlComparison.java#L138

Reply:
Okay, since two of the overloaded `compareInstances` methods have the default predicate shown, I've just made reference to those methods in the JavaDocs: 

https://github.com/reactome/data-release-pipeline/pull/120/commits/7dc226d5925544de4ad351a92a286ed63752f0e1